umapi_client/functional.py

```diff
@@ -329,39 +317,32 @@ def maybe_split_groups(self, max_groups):
         new_commands = []
```

Comment:
OK, so here is a really nit-worthy comment: I think you should interchange the use of the var names `new_commands` and `split_commands` in this function.  You should have `split_commands` be the top-level collection of commands that will replace the existing ones, consistent with the function name.  And you should have `new_commands` be the inner-loop accumulator of `new_command` values, because those are the new commands, after all...